### ISSUE TITLE
Swagger Codegen can't use my own template files

### ISSUE THREAD

**Issue by parhamahmady** [swagger-api/swagger-codegen#11621]
Hello

When  I set template directory for swagger-codegen-maven-plugin to generate java code from my own template files.
the build fails.
I use this configuration for maven plugin:

                <plugin>
                    <groupId>io.swagger.codegen.v3</groupId>
                    <artifactId>swagger-codegen-maven-plugin</artifactId>
                    <version>3.0.30</version>
                    <executions>
                        <execution>
                            <goals>
                                <goal>generate</goal>
                            </goals>
                            <configuration>
                                <inputSpec>${project.basedir}/src/main/resources/openapi.json</inputSpec>
                                <language>spring</language>
                                <library>spring-cloud</library>
                                <templateDirectory>src/main/resources/JavaSpring</templateDirectory>
                                <configOptions>
                                    <generateForOpenFeign>true</generateForOpenFeign>
                                    <groupId>${project.parent.groupId}</groupId>
                                    <artifactId>${project.name}-java-client</artifactId>
                                    <artifactVersion>${project.version}</artifactVersion>
                                    <dateLibrary>java8</dateLibrary>
                                </configOptions>
                            </configuration>
                        </execution>
                    </executions>
                </plugin>

The plugin can't find my files :
java.io.FileNotFoundException: /handlebars/JavaSpring/C:/Users/myname/Desktop/projects/test-project-api/src/main/resources/JavaSpring/model.mustache

I was wondering if the problem caused by wrong prefix of URL ?

**Comment by laura-rodriguez** [comment]
I'm facing the same issue using the CLI. It seems the prefix `/handlebars/<lang>` is always added regardless of indicating the templateDir path. Note `Caused by: java.io.FileNotFoundException: /handlebars/csharp/C:/source/oas3-migration/dotnet-swagger-templates/model.mustache` in the following error log:

```
java --add-opens=java.base/java.util=ALL-UNNAMED -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -l csharp -i C:/spec.json -o /tmp/dotnet -t C:\source\oas3-migration\dotnet-swagger-templates

15:00:42.525 [Thread-0] INFO  i.s.c.v.g.dotnet.CSharpClientCodegen - Generating code for .NET Framework v4.5
Exception in thread "Thread-0" java.lang.RuntimeException: Could not generate model 'AccessPolicy'
        at io.swagger.codegen.v3.DefaultGenerator.generateModels(DefaultGenerator.java:451)
        at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:779)
        at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:388)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.FileNotFoundException: /handlebars/csharp/C:/source/oas3-migration/dotnet-swagger-templates/model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:438)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:419)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getHandlebars(HandlebarTemplateEngine.java:44)
        at io.swagger.codegen.v3.templates.HandlebarTemplateEngine.getRendered(HandlebarTemplateEngine.java:24)
        at io.swagger.codegen.v3.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:795)
        at io.swagger.codegen.v3.DefaultGenerator.generateModels(DefaultGenerator.java:438)
        ... 3 more
```

**Comment by HugoMario** [comment]
fixed by #11689
